client/app.js

```diff
@@ -148,6 +148,183 @@ tie.constant('PREREQ_CHECK_TYPE_BAD_IMPORT', 'badImport');
  * @constant
  */
 tie.constant('PREREQ_CHECK_TYPE_GLOBAL_CODE', 'globalCode');
+tie.constant('PREREQ_CHECK_TYPE_WRONG_LANG', 'wrongLang');
+
+/**
+ * Dictionary of wrong language detection errors and their related information
+ *
+ * @type {{}}
+ */
+tie.constant('WRONG_LANGUAGE_ERRORS', {
```

Comment:
I'd suggest creating a new appSpec.js file.

(Also, if you have time, it would be nice to do something similar for the other constant.)